### ISSUE TITLE
Broken private Vimeo embedded download

### ISSUE THREAD

**Issue by gabrieleancora** [yt-dlp/yt-dlp#2128]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.25**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Everywhere

### Description

Hi!
I'm trying to download a private embedded Vimeo video from a website, and some time ago I found that the easiest way to do it from an unsupported site was to open the console, wait for the master.json to appear and copy it's whole url from it replacing the .json part with .mpd.
It has always worked pretty well (and it still works on Youtube-dl), but when I try to use that method with yt-dlp, it doesn't work.

The video download is fine, but when it has to download the audio part, i get a 403 error from vimeo.

As you can notice, I have to put the `--no-check-certificates` parameter to get this error, because otherwise I get an hostname mismatch error, but I don't think it's the main issue here.

I _think_ both errors are caused by an incorrect URL parsing, because I noticed that the audio URL that is trying to download from is different from the one extracted at the beginning, with a lot of `../` added to it.

If you want, for reference, here it is a (successful) download for the same video using youtube-dl (I know that the url isn't the same, but I had to use a different one because the first url expired and vimeo was giving me a second, newer, one): https://pastebin.com/9nuv50WK

### Verbose log

```shell
[debug] Command-line config: ['--no-check-certificates', '-Uv', 'https://24vod-adaptive.akamaized.net/exp=1640538410~acl=%2Ff570a4ba-ef7f-4a81-90ea-a58ca89a4d0f%2F%2A~hmac=a851cfd40ffbed5685a2710b1f483737aa2facd238e43e0185ac78f6deb82c6a/f570a4ba-ef7f-4a81-90ea-a58ca89a4d0f/sep/video/03ec5e67,0365e3b7,9feccfed,e8bee4d6/audio/4139dffb,06986cf0/master.mpd', '-o', 'TEST.mp4']
[debug] User config file: /home/gabriele/.config/yt-dlp/config
[debug] User config: ['-o', '%(title)s.%(ext)s']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.25 [87e049962] (zip)
[debug] Python version 3.8.10 (CPython 64bit) - Linux-4.4.0-19041-Microsoft-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4
[debug] Optional libraries: keyring, sqlite
[debug] Proxy map: {}
Latest version: 2021.12.25, Current version: 2021.12.25
yt-dlp is up to date (2021.12.25)
[debug] [generic] Extracting URL: https://24vod-adaptive.akamaized.net/exp=1640538410~acl=%2Ff570a4ba-ef7f-4a81-90ea-a58ca89a4d0f%2F%2A~hmac=a851cfd40ffbed5685a2710b1f483737aa2facd238e43e0185ac78f6deb82c6a/f570a4ba-ef7f-4a81-90ea-a58ca89a4d0f/sep/video/03ec5e67,0365e3b7,9feccfed,e8bee4d6/audio/4139dffb,06986cf0/master.mpd
[generic] master: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] master: Downloading webpage
[generic] master: Extracting information
[debug] Identified a DASH manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] master: Downloading 1 format(s): video-e8bee4d6+audio-4139dffb
[debug] Invoking downloader on "https://24vod-adaptive.akamaized.net/exp=1640538410~acl=%2Ff570a4ba-ef7f-4a81-90ea-a58ca89a4d0f%2F%2A~hmac=a851cfd40ffbed5685a2710b1f483737aa2facd238e43e0185ac78f6deb82c6a/f570a4ba-ef7f-4a81-90ea-a58ca89a4d0f/sep/video/03ec5e67,0365e3b7,9feccfed,e8bee4d6/audio/4139dffb,06986cf0/master.mpd"
[dashsegments] Total fragments: 67
[download] Destination: TEST.fvideo-e8bee4d6.mp4
[download] 100% of 127.13MiB in 01:05
[debug] Invoking downloader on "https://24vod-adaptive.akamaized.net/exp=1640538410~acl=%2Ff570a4ba-ef7f-4a81-90ea-a58ca89a4d0f%2F%2A~hmac=a851cfd40ffbed5685a2710b1f483737aa2facd238e43e0185ac78f6deb82c6a/f570a4ba-ef7f-4a81-90ea-a58ca89a4d0f/sep/video/03ec5e67,0365e3b7,9feccfed,e8bee4d6/audio/4139dffb,06986cf0/../../../../../parcel/audio/4139dffb.mp4"
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2971, in process_info
    partial_success, real_download = self.dl(fname, new_info)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2699, in dl
    return fd.download(name, new_info, subtitle)
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/common.py", line 440, in download
    ret = self.real_download(filename, info_dict)
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/http.py", line 371, in real_download
    establish_connection()
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/http.py", line 119, in establish_connection
    raise err
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/http.py", line 113, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3456, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 563, in error
    result = self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 755, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

**Comment by pukkandan** [comment]
The link expires so we can't test. Give link to the actual website

**Comment by gabrieleancora** [comment]
Yeah sorry, the website is: https://lovelive-aqoursclub.jp/ , and the videos are in the "Challenge" section.

But as you can see, you need to login. How can I give the login data to you safely?

**Comment by nao20010128nao** [comment]
Give cookies to yt-dlp and feed the URL where Vimeo embed appears

**Comment by gabrieleancora** [comment]
If I try to download the video using the URL of the page itself using the cookies, i get this:

```shell
[debug] Command-line config: ['-Uv', '--cookies', 'cookies.txt', 'https://lovelive-aqoursclub.jp/mob/cont/contLis.php?site=AC&ima=3402&cd=180&ct=CHALLENGE']
[debug] User config file: /home/gabriele/.config/yt-dlp/config
[debug] User config: ['-o', '%(title)s.%(ext)s']
WARNING: skipping cookie file entry due to invalid expires at 1.7976931348623157e+308: '.fandom.com\tTRUE\t/\tFALSE\t1.7976931348623157e+308\tfandom-desktop-banner-seen\t1\n'
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8] (zip)
[debug] Python version 3.8.10 (CPython 64bit) - Linux-4.4.0-19041-Microsoft-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4
[debug] Optional libraries: secretstorage, sqlite
[debug] Proxy map: {}
Latest version: 2021.12.27, Current version: 2021.12.27
yt-dlp is up to date (2021.12.27)
[debug] [generic] Extracting URL: https://lovelive-aqoursclub.jp/mob/cont/contLis.php?site=AC&ima=3402&cd=180&ct=CHALLENGE
[generic] contLis: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] contLis: Downloading webpage
[generic] contLis: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://lovelive-aqoursclub.jp/mob/cont/contLis.php?site=AC&ima=3402&cd=180&ct=CHALLENGE
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1353, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1423, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/generic.py", line 3829, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://lovelive-aqoursclub.jp/mob/cont/contLis.php?site=AC&ima=3402&cd=180&ct=CHALLENGE
```

The page is a list of videos and each time you press one, a little js popup appears (like a photo gallery) and inside it there is the embedded vimeo video using an iframe.

Obviously if I try downloading using the iframe url itself i get this error:  
```shell
ERROR: [vimeo] 659589967: Cannot download embed-only video without embedding URL. Please call yt-dlp with the URL of the page that embeds this video.
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/vimeo.py", line 791, in _real_extract
    raise ExtractorError(
```

Ps. @nao20010128nao  I purchased the cheaper version, [this one](https://www.amazon.co.jp/product/dp/B08ZJH55C3/). Both versions have the code to access the Aqours club for one year.

**Comment by pukkandan** [comment]
Try passing the video URL instead with referer to this site (and cookies ofc). Like this
```
yt-dlp -Uv http://player.vimeo.com/video/54469442 --referer "https://lovelive-aqoursclub.jp/mob/cont/contLis.php?site=AC&ima=3402&cd=180&ct=CHALLENGE" --cookies cookies.txt
```

If that still doesn't work, I can take a look if you can share account details. You can reach me on discord/reddit or over email

**Comment by gabrieleancora** [comment]
With those options it works fine, without having to bypass the certificate check.

```shell
[debug] Command-line config: ['-Uv', 'https://player.vimeo.com/video/659589967', '--referer', 'https://lovelive-aqoursclub.jp/mob/cont/contLis.php?site=AC&ima=0327&cd=180&ct=CHALLENGE', '--cookies', 'cookies.txt']
WARNING: skipping cookie file entry due to invalid expires at 1.7976931348623157e+308: '.fandom.com\tTRUE\t/\tFALSE\t1.7976931348623157e+308\tfandom-desktop-banner-seen\t1\n'
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.27 [6223f67a8] (zip)
[debug] Python version 3.8.10 (CPython 64bit) - Linux-4.4.0-19041-Microsoft-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4
[debug] Optional libraries: secretstorage, sqlite
[debug] Proxy map: {}
Latest version: 2021.12.27, Current version: 2021.12.27
yt-dlp is up to date (2021.12.27)
[debug] [vimeo] Extracting URL: https://player.vimeo.com/video/659589967
[vimeo] 659589967: Downloading webpage
[vimeo] 659589967: Downloading akfire_interconnect_quic m3u8 information
[vimeo] 659589967: Downloading akfire_interconnect_quic m3u8 information
[vimeo] 659589967: Downloading akfire_interconnect_quic MPD information
[vimeo] 659589967: Downloading akfire_interconnect_quic MPD information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, lang, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 659589967: Downloading 1 format(s): hls-akfire_interconnect_quic-2937
[debug] Invoking downloader on "https://24vod-adaptive.akamaized.net/exp=1640632269~acl=%2Ff570a4ba-ef7f-4a81-90ea-a58ca89a4d0f%2F%2A~hmac=d39c7ba3fa58e509364a209325272cff5754e7dbad6eaa5b549c42ed0672f5db/f570a4ba-ef7f-4a81-90ea-a58ca89a4d0f/video/e8bee4d6/playlist.m3u8?query_string_ranges=1"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 66
[download] Destination: #37_箱の中身を探る冒険！ [659589967].mp4
[download] 100% of 143.80MiB in 01:38
[debug] ffprobe command line: ffprobe -hide_banner -show_format -show_streams -print_format json 'file:#37_箱の中身を探る冒険！ [659589967].mp4'
[FixupM3u8] Fixing MPEG-TS in MP4 container of "#37_箱の中身を探る冒険！ [659589967].mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:#37_箱の中身を探る冒険！ [659589967].mp4' -c copy -map 0 -dn -f mp4 -bsf:a aac_adtstoasc -movflags +faststart 'file:#37_箱の中身を探る冒険！ [659589967].temp.mp4'
```

Do you still want the account details or we just mark that as the solution?

**Comment by pukkandan** [comment]
Only if you still want us to check why your original command was failing. If not, u can close this

**Comment by doradod** [comment]
Hi, I'm trying to do something like this

yt-dlp -Uv https://player.vimeo.com/video/826375357 --referer "https://circle.digitalambition.co" --cookies cookies.txt

[debug] Command-line config: ['-Uv', 'https://player.vimeo.com/video/826375357', '--referer', 'https://circle.digitalambition.co', '--cookies', 'cookies.txt']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.07 [377e85a17] (pip)
[debug] Python 3.11.6 (CPython arm64 64bit) - macOS-14.0-arm64-arm-64bit (OpenSSL 3.1.3 19 Sep 2023)
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2023.07.22, mutagen-1.47.0, sqlite3-3.43.2, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1886 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.10.07, Current version: stable@2023.10.07
yt-dlp is up to date (stable@2023.10.07)
[vimeo] Extracting URL: https://player.vimeo.com/video/826375357
[vimeo] 826375357: Downloading webpage
ERROR: [vimeo] 826375357: Unable to download webpage: HTTP Error 404: Not Found (caused by <HTTPError 404: Not Found>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/extractor/vimeo.py", line 828, in _real_extract
    webpage, urlh = self._download_webpage_handle(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 903, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 860, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/networking/_urllib.py", line 410, in _send
    res = opener.open(urllib_req, timeout=float(request.extensions.get('timeout') or self.timeout))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 525, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 634, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 563, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 4051, in urlopen
    return self._request_director.send(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/networking/common.py", line 114, in send
    response = handler.send(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/networking/_helper.py", line 204, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/networking/common.py", line 325, in send
    return self._send(request)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/networking/_urllib.py", line 415, in _send
    raise HTTPError(UrllibResponseAdapter(e.fp), redirect_loop='redirect error' in str(e)) from e
yt_dlp.networking.exceptions.HTTPError: HTTP Error 404: Not Found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 847, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/yt-dlp/2023.10.7/libexec/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 4070, in urlopen
    raise _CompatHTTPError(e) from e
yt_dlp.networking.exceptions._CompatHTTPError: HTTP Error 404: Not Found

Any idea if i need to do something different?

**Comment by bashonly** [comment]
https://player.vimeo.com/video/826375357

> Sorry
> This video does not exist.
